cmake/AddToTarget.cmake

```diff
@@ -0,0 +1,39 @@
+if(CELERITY_SYCL_IMPL STREQUAL "DPC++")
+  function(add_sycl_to_target)
+    set(options)
+    set(one_value_args TARGET)
+    set(multi_value_args)
+    cmake_parse_arguments(ADD_SYCL
+      "${options}"
+      "${one_value_args}"
+      "${multi_value_args}"
+      ${ARGN}
+      )
+    # Wno-deprecated-declarations: Kernel offsets are used all over the place
+    set(DPCPP_FLAGS "-fsycl;-sycl-std=2020;-fsycl-targets=${CELERITY_DPCPP_TARGETS};-DCELERITY_DPCPP=1;-Wno-deprecated-declarations;${DPCPP_FLAGS}")
```

Comment:
DPC++ supports a macro `SYCL2020_DISABLE_DEPRECATION_WARNINGS` - should we be using that instead..? 🤔

Reply:
Actually, we don't need to suppress warnings at all anymore (see the comment!)

Reply:
Even better!